app/api/route/message_api.rb

```diff
@@ -0,0 +1,31 @@
+module Route
+  class MessageApi < Base
+    paginate paginate_settings
+
+    params do
+      optional :thread, type: Integer, desc: 'ID of thread'
+      optional :order_by, type: Symbol, values:%i( created_at  ), desc: 'Order of returned issues.'
+      optional :order, type: Symbol, values:%i(asc desc), default: :asc
+      optional :submit_external, type: String, desc: 'Filter by external service submission mark'
+    end
+
+    helpers do
+      def post_or_get_messages
+        scope = Message.joins(:thread).includes(:created_by)
+        scope = scope.where('message_threads.privacy' => 'public')
+        scope = scope.where(censored_at: nil)
+        scope = scope.order(params[:order_by] => params[:order]) if params[:order_by]
+        scope = scope.where(thread_id: params[:thread]) if params[:thread]
+        scope = scope.where('message_threads.submit_external' => params[:submit_external]) if params[:submit_external]
+        scope = paginate scope
```

Comment:
You've not `def as_json` on the `Message` so we will be sending all the fields that exist in the db, like `check_reason` and `public_token`.  Can we trim it to a specific white list please?
e.g.
https://github.com/cyclestreets/cyclescape/blob/staging/app/decorators/library/item_decorator.rb#L20-L28

Reply:
+1